yapapi/event_dispatcher.py

```diff
@@ -1,27 +1,29 @@
 import asyncio
-from typing import Callable, List
+from typing import Callable, Dict, Set, Type
 
 from yapapi import events
 from yapapi.utils import AsyncWrapper
 
 
 class AsyncEventDispatcher:
     def __init__(self):
-        self._consumers: List[AsyncWrapper] = []
+        self._consumers: Dict[AsyncWrapper, Set[Type[events.Event]]] = {}
```

Comment:
I think a tuple of (AsyncWrapper, Set) would be more appropriate here... a Dict seems strange in this usage... especially with keys being Callables ...

Reply:
Keys are objects, not Callables - but this changes nothing.

I think the Dict seems strange because there are a lot of languages that disallow this. But `python` allows, works well & also it's very readable. I'll be happy to change my mind (and also the code) if this is a common objection.

Reply:
I did a quick googling and I don't see why class instances shouldn't be dictionary keys.
They are hashable, they have unique never-changing `id()` - I don't see a problem here. The same goes for Callables.